lager/detail/nodes.hpp

```diff
@@ -155,17 +148,18 @@ class reader_node : public reader_node_base
     void push_down(U&& value)
     {
         if (has_changed(value, current_)) {
-            current_ = std::forward<U>(value);
-            flags_ |= needs_send_down_flag;
+            current_         = std::forward<U>(value);
+            needs_send_down_ = true;
         }
     }
 
     void send_down() final
     {
         recompute();
-        if (flags_ & needs_send_down_flag) {
-            last_  = current_;
-            flags_ = needs_notify_flag;
+        if (needs_send_down_) {
+            last_            = current_;
+            needs_send_down_ = false;
+            needs_notify_    = true;
```

Comment:
So before we where setting both needs_send_down and neets_notify in one step and now in two. I do not think that this is an improvement. (This is actually moving the code back to how it was some months ago...)

Reply:
ok, but adding `outermost` to the `flags_` variable means that we now need to do some additional bitwise masking to avoid touching it.

I feel like I just need to write an extra struct to encapsulate the bitwise stuff and expose it via a more readable interface...

Reply:
Nah ok.